v2/lints/cabf_br/lint_ocsp_id_pkix_ocsp_nocheck_ext_not_included_non_server_auth_test.go

```diff
@@ -0,0 +1,185 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/test"
+)
+
+func TestOCSPIDPKIXOCSPNocheckExtNotIncludedNonServerAuth(t *testing.T) {
+	testCases := []struct {
+		Name           string
+		Filename       string
+		ExpectedResult lint.LintStatus
+	}{
+
+		// Legend for the nameing:
+		// o1 --> EKU OCSPSigning set; o0 not set
+		// s1 --> EKU serverAuth set, s0 not set
+		// ep1 --> EKU emailProtection set, ep0 not set
+		// a1 --> EKU anyExtendedKeyUsage set, a0 not set
+		// nc1 --> noCheck set, nc0 not set
```

Comment:
I don't know that this combinatorial approach makes sense anymore, now that you've fundamentally redesigned the thing.

As a reviewer, these fundamental redesigns make it difficult to pick up, because it means I have to start from scratch all over again, and that's why I was hoping we might take a step back here, make sure we're in agreement and understand the challenges, and figure out forward. Reviewing lots of code, only for it to be thrown away, I'm sure is not a good use of your time, but it also doesn't feel good as a reviewer.

Does it make sense to take a step back, discuss on the issue the possible approaches and pro/cons you see, and make sure we're in agreement? Given that this is now the... I believe fourth... different approach, it seems like you've thought a lot about this, and it might help to summarize that clearly and their merits, and then we can easily review the implementation as aligned with whatever tradeoffs we accept.

For example, my original suggestion was only to have 5 tests, and so I'm not sure why you've covered this extensive permutation here of all combinations. It's certainly interesting, but I also worry it's not really scalable or manageable for all these certificates, and may not be buying us anything for the conditions we logically care about. I'm happy, of course, to better understand and learn more, because I'm sure there's a good reason you thought this would be better, and it's just not obvious to me.